lib/gpu/command_buffer.cc

```diff
@@ -34,18 +34,32 @@ void CommandBuffer::AddRenderPass(
 }
 
 bool CommandBuffer::Submit() {
-  for (auto& encodable : encodables_) {
-    encodable->EncodeCommands();
-  }
-
-  return context_->GetCommandQueue()->Submit({command_buffer_}).ok();
+  return CommandBuffer::Submit({});
 }
 
 bool CommandBuffer::Submit(
     const impeller::CommandBuffer::CompletionCallback& completion_callback) {
   for (auto& encodable : encodables_) {
     encodable->EncodeCommands();
   }
+
+  if (context_->GetBackendType() == impeller::Context::BackendType::kOpenGLES) {
```

Comment:
I thought the whole point of the GLES backend's task reactor design was that we could call these methods from any thread?

Reply:
blocking on the raster task runner for like this seems pretty bad, you're essentially de-pipeling .